### PR TITLE
Print warning when running on newer version of Node.js

### PR DESCRIPTION
**Pull request by LinusU**
Right now, both Node.js 13 and 14 is "Current release"

Screenshot 2020-04-25 at 13 25 13

### CODE REVIEW

**Comment by brentvatne** [comment]
we haven't tried using it with node 14 yet, perhaps we should add a new class of warning that this is a new release and there may be issues, but let people proceed with using expo-cli with it. can you update the pr to do that?

**Comment by LinusU** [comment]
@brentvatne does this look good?

btw, `chalk` doesn't even import on Node.js <8, so effectively the error is only printed for 8.x, 11.x.  Maybe that isn't a problem though? I'm not really sure what the motivation behind the warning is ☺️

File: packages/expo-cli/bin/expo.js [hunk 1]
```diff
@@ -13,23 +13,29 @@ const isSupported = supportedVersions.some(function(supported) {
...
 
-  const maxSupportedVersion = supportedVersions[supportedVersions.length - 1];
-  const versionIsHigherThanSupported = semver.gtr(version, maxSupportedVersion.range);
+const maxSupportedVersion = supportedVersions[supportedVersions.length - 1];
+const versionIsHigherThanSupported = semver.gtr(version, maxSupportedVersion.range);
 
+if (versionIsHigherThanSupported) {
+  console.error(
+    chalk.yellow(
+      `WARNING: Node.js version ${version} has not yet been tested.\n\n` +
```
Comment (brentvatne) [inline packages/expo-cli/bin/expo.js:26]:
```suggestion
      `WARNING: expo-cli has not yet been tested against Node.js version ${version}. If you encounter any issues, please report them to https://github.com/expo/expo-cli/issues\n\n` +
```

**Comment by brentvatne** [comment]
@LinusU - i didn't know that :O

the purpose of this is that we don't want to have people creating issues about problems they encounter with unsupported versions of expo-cli, and we found that if we don't block it from starting people will just ignore the warnings and try anyways. we can be pretty sure that on node < 10 people are going to have problems, so we just save them the trouble of encountering an obscure error and stop them from starting it at all.

thanks for the pr!

**Comment by LinusU** [comment]
> the purpose of this is that we don't want to have people creating issues about problems they encounter with unsupported versions of expo-cli, and we found that if we don't block it from starting people will just ignore the warnings and try anyways.

Thanks for the explanation, that makes sense!

> i didn't know that :O

`semver` also only supports Node.js 10 and up, so we probably need to do this without dependencies if you want it to work on any old version. We are doing something similar in [StandardJS here](https://github.com/standard/standard/blob/37d8bb202bf7f250f7d403c4c35d0bd097d6b39f/bin/cmd.js). I'd be happy to submit a PR if you want to...

**Comment by brentvatne** [comment]
that would be great, thank you @LinusU!

**Comment by LinusU** [comment]
Thanks for merging! Follow up PR is here: #2030

**Pull request merged**